src/game_events/wmi_manager.cpp

```diff
@@ -97,8 +95,17 @@ bool wmi_manager::fire_item(
 	if(wmi->can_show(hex, gamedata, fc)) {
 		wmi->fire_event(hex, gamedata);
 	}
+
+	// Restore old values
 	gamedata.get_variable("x1") = x1;
 	gamedata.get_variable("y1") = y1;
+
+	// Update toggle variable, if applicable
+	if(wmi->is_toggle_item()) {
+		auto& state = gamedata.get_variable(wmi->toggle_state_variable());
+		state = !state.to_bool();
+	}
+
```

Comment:
i think setting the variable before the wml code runs would make it easier to use the updated variable in the wml code

Reply:
Agreed, done.